packages/core/src/components/graphs/bar-stacked.ts

```diff
@@ -123,30 +127,76 @@ export class StackedBar extends Bar {
+					return this.services.axes.getXValue(d[1], i) - length;
 				})
-				.attr("width", this.getBarWidth.bind(this))
-				.transition(this.services.transitions.getTransition("bar-update-enter", animate))
-				.attr("y", (d, i) => this.services.axes.getYValue(d[1], i))
-				.attr("fill", d => this.model.getFillScale()[d.datasetLabel](d.label))
-				.attr("height", (d, i) => {
+				.attr("width", (d, i) => {
+					if (xScale.scaleType === ScaleTypes.LABELS ) {
+						length = Math.abs(this.services.axes.getXValue(d.label, i) - xScale.scale.range()[0]);
+					} else {
+						length = Math.abs(this.services.axes.getXValue(d[0]) - this.services.axes.getXValue(d[1]));
+					}
 					const { datasetLabel } = d;
 					const datasetLabelIndex = stackKeys.indexOf(datasetLabel);
-					const height = this.services.axes.getYValue(d[0]) - this.services.axes.getYValue(d[1]);
-
-					if (datasetLabelIndex > 0 && height >= options.bars.dividerSize) {
-						return height - options.bars.dividerSize;
+					// create dividers between every bar
+					if (datasetLabelIndex < (datasetLabel.length - 1) && length >= options.bars.dividerSize) {
+						return length - options.bars.dividerSize;
 					}
-
-					return height;
-				})
+					return length;
+				} )
+				.transition(this.services.transitions.getTransition("bar-update-enter", animate))
+				.attr("y", (d, i) => { return this.services.axes.getYValue(d, i) - this.getBarWidth() / 2; })
+				.attr("fill", d => this.model.getFillScale()[d.datasetLabel](d.label))
+				.attr("height", this.getBarWidth.bind(this))
 				.attr("opacity", 1);
+			} else {
+				// vertical stacked bar code
+				const yScale = this.services.axes.getMainYAxis();
+				bars.enter()
+					.append("rect")
+					.merge(bars)
+						.classed("bar", true)
+						.attr("x", (d, i) => {
+							const barWidth = this.getBarWidth();
+							return this.services.axes.getXValue(d, i) - (barWidth / 2);
+						})
+						.attr("width", this.getBarWidth.bind(this))
+						.transition(this.services.transitions.getTransition("bar-update-enter", animate))
+						.attr("y", (d, i) => this.services.axes.getYValue(d[1], i))
+						.attr("fill", d => this.model.getFillScale()[d.datasetLabel](d.label))
+						.attr("height", (d, i) => {
+							const { datasetLabel } = d;
+							const datasetLabelIndex = stackKeys.indexOf(datasetLabel);
+							let height;
+							// determine height based on the y axis
+							if (yScale.scaleType === ScaleTypes.LABELS ) {
+								height = Math.abs(yScale.scale.range()[0] - this.services.axes.getYValue(d.label, i));
+							} else {
+								height = this.services.axes.getYValue(d[0]) - this.services.axes.getYValue(d[1]);
+							}
+							// create dividers between bars
+							if (datasetLabelIndex > 0 && height >= options.bars.dividerSize) {
+								return height - options.bars.dividerSize;
+							}
+
+							return height;
+						})
+						.attr("opacity", 1);
+			}
```

Comment:
Same note about code duplication here ... not sure _how_ to fix it, but it'd be nice if we could